sklearn/metrics/tests/test_pairwise.py

```diff
@@ -1294,3 +1295,45 @@ def test_pairwise_distances_data_derived_params(n_jobs, metric, dist_function,
 
         assert_allclose(dist, expected_dist_explicit_params)
         assert_allclose(dist, expected_dist_default_params)
+
+
+@pytest.mark.parametrize(
+        'metric', [
+            'braycurtis', 'canberra', 'chebyshev',
+            'correlation', 'hamming', 'mahalanobis', 'minkowski', 'seuclidean',
+            'sqeuclidean', 'cityblock', 'cosine', 'euclidean'])
+@pytest.mark.parametrize(
+        "dtype",
+        [np.float32, np.float64])
+@pytest.mark.parametrize("y_is_x", [True, False], ids=["Y is X", "Y is not X"])
+def test_numeric_pairwise_distances_datatypes(metric, dtype, y_is_x):
+    # Check that pairwise distances gives the same result as pdist and cdist
+    # regardless of input datatype when using any scipy metric for comparing
+    # numeric vectors
+    #
+    # This test is necessary because pairwise_distances used to throw an
+    # error when using metric='seuclidean' and the input data was not
+    # of type np.double (#15730)
+    #
+    # This test has introduced a few xfails on the Windows test benches
+    # because 'array type float64 is unsupported in linalg'.
+    if (metric == 'mahalanobis' and
+            dtype == np.longdouble and
+            platform.startswith('win')):
+        pytest.xfail('array type float64 is unsupported in ' +
+                     'linalg for this testbench')
```

Comment:
Without np.float128, this should no longer be needed

Reply:
fixed in d9a632521d10ef74a153130bd43ceb1e6f99add4 and aec9405883d6da7cda25ec70ba8bcce4cd37b520